src/wazuh_db/wdb_agents.c

```diff
@@ -15,31 +15,129 @@ int wdb_agents_insert_vuln_cve(wdb_t *wdb, const char* name, const char* version
     return wdb_exec_stmt_silent(stmt);
 }
 
-int wdb_agents_clear_vuln_cve(wdb_t *wdb) {
+int wdb_agents_update_status_vuln_cves(wdb_t *wdb, const char* old_status, const char* new_status) {
+
+    bool update_all = (strcmp(old_status, "*") == 0);
+
+    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, update_all ? WDB_STMT_VULN_CVES_UPDATE_ALL : WDB_STMT_VULN_CVES_UPDATE);
 
-    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_VULN_CVE_CLEAR);
     if (stmt == NULL) {
         return OS_INVALID;
     }
 
+    if (update_all) {
+        sqlite3_bind_text(stmt, 1, new_status, -1, NULL);
+    } else {
+        sqlite3_bind_text(stmt, 1, new_status, -1, NULL);
+        sqlite3_bind_text(stmt, 2, old_status, -1, NULL);
+    }
+
     return wdb_exec_stmt_silent(stmt);
 }
 
-int wdb_agents_update_status_vuln_cve(wdb_t *wdb, const char* old_status, const char* new_status) {
-    
-    bool update_all = (strcmp(old_status, "*") == 0);
+int wdb_agents_remove_vuln_cves(wdb_t *wdb, const char* cve, const char* reference) {
+    if (!cve || !reference) {
+        mdebug1("Invalid data provided");
+        return OS_INVALID;
+    }
+
+    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_VULN_CVES_DELETE_ENTRY);
 
-    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, update_all ? WDB_STMT_VULN_CVE_UPDATE_ALL : WDB_STMT_VULN_CVE_UPDATE);
-    
     if (stmt == NULL) {
         return OS_INVALID;
     }
 
-    if (update_all) {
-        sqlite3_bind_text(stmt, 1, new_status, -1, NULL);
-    } else {
-        sqlite3_bind_text(stmt, 1, new_status, -1, NULL);
-        sqlite3_bind_text(stmt, 2, old_status, -1, NULL);
+    sqlite3_bind_text(stmt, 1, cve, -1, NULL);
+    sqlite3_bind_text(stmt, 2, reference, -1, NULL);
+
+    return wdb_exec_stmt_silent(stmt);
+}
+
+wdbc_result wdb_agents_remove_by_status_vuln_cves(wdb_t *wdb, const char* status, char **output) {
+    sqlite3_stmt* stmt = NULL;
+    unsigned response_size = 2; //Starts with "[]" size
+    wdbc_result wdb_res = WDBC_UNKNOWN;
+
+    os_calloc(WDB_MAX_RESPONSE_SIZE, sizeof(char), *output);
+    char *response_aux = *output;
+
+    //Prepare SQL query
+    if (stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_VULN_CVES_SELECT_BY_STATUS), !stmt) {
+        mdebug1("Cannot cache statement");
```

Comment:
The **wdb_init_stmt_in_cache()** method already has a mdebug1 message with `"Cannot cache statement`. Wouldn't it be a repetition ?

Reply:
Totally right. Fixed.